addons/sourcemod/scripting/mgemod.sp

```diff
@@ -2279,128 +2295,128 @@ LoadSpawnPoints()
 	}
 }
 
-ResetPlayer(client)
+int ResetPlayer(int client)
 {
-	new arena_index = g_iPlayerArena[client];
-	new player_slot = g_iPlayerSlot[client];
+	int arena_index = g_iPlayerArena[client];
+	int player_slot = g_iPlayerSlot[client];
 	
 	
 	if (!arena_index || !player_slot)
 	{
 		return 0;
 	}
-
+	
 	g_iPlayerSpecTarget[client] = 0;
 	
-	if(player_slot == SLOT_ONE || player_slot == SLOT_THREE)
+	if (player_slot == SLOT_ONE || player_slot == SLOT_THREE)
 		ChangeClientTeam(client, TEAM_RED);
 	else
 		ChangeClientTeam(client, TEAM_BLU);
-		
+	
 	//This logic doesn't work with 2v2's
 	//new team = GetClientTeam(client);
 	//if (player_slot - team != SLOT_ONE - TEAM_RED) 
 	//	ChangeClientTeam(client, player_slot + TEAM_RED - SLOT_ONE);
 	
-
-	new TFClassType:class;
+	TFClassType class;
 	class = g_tfctPlayerClass[client] ? g_tfctPlayerClass[client] : TFClass_Soldier;
-
+	
 	if (!IsPlayerAlive(client) || g_bArenaBBall[arena_index])
 	{
-		if (class != TF2_GetPlayerClass(client)) 
-			TF2_SetPlayerClass(client,class);
-
+		if (class != TF2_GetPlayerClass(client))
+			TF2_SetPlayerClass(client, class);
+		
 		TF2_RespawnPlayer(client);
 	} else {
 		TF2_RegeneratePlayer(client);
 		ExtinguishEntity(client);
 	}
-
+	
 	g_iPlayerMaxHP[client] = GetEntProp(client, Prop_Data, "m_iMaxHealth");
-
+	
 	if (g_bArenaMidair[arena_index])
 		g_iPlayerHP[client] = g_iMidairHP;
 	else
-		g_iPlayerHP[client] = g_iPlayerHandicap[client] ? g_iPlayerHandicap[client] : RoundToNearest(float(g_iPlayerMaxHP[client])*g_fArenaHPRatio[arena_index]);
-
+		g_iPlayerHP[client] = g_iPlayerHandicap[client] ? g_iPlayerHandicap[client] : RoundToNearest(float(g_iPlayerMaxHP[client]) * g_fArenaHPRatio[arena_index]);
+	
 	if (g_bArenaMGE[arena_index] || g_bArenaBBall[arena_index])
-		SetEntProp(client, Prop_Data, "m_iHealth", g_iPlayerHandicap[client] ? g_iPlayerHandicap[client] : RoundToNearest(float(g_iPlayerMaxHP[client])*g_fArenaHPRatio[arena_index]));
-
+		SetEntProp(client, Prop_Data, "m_iHealth", g_iPlayerHandicap[client] ? g_iPlayerHandicap[client] : RoundToNearest(float(g_iPlayerMaxHP[client]) * g_fArenaHPRatio[arena_index]));
+	
 	ShowPlayerHud(client);
 	ResetClientAmmoCounts(client);
-	CreateTimer(0.1,Timer_Tele,GetClientUserId(client));
-
+	CreateTimer(0.1, Timer_Tele, GetClientUserId(client));
+	
 	return 1;
 }
 
-ResetKiller(killer,arena_index)
-{
-	TF2_RegeneratePlayer(killer);
-	new reset_hp = g_iPlayerHandicap[killer] ? g_iPlayerHandicap[killer] : RoundToNearest(float(g_iPlayerMaxHP[killer])*g_fArenaHPRatio[arena_index]);
+void ResetKiller(int killer, int arena_index)
+{	
+	int reset_hp = g_iPlayerHandicap[killer] ? g_iPlayerHandicap[killer] : RoundToNearest(float(g_iPlayerMaxHP[killer]) * g_fArenaHPRatio[arena_index]);
 	g_iPlayerHP[killer] = reset_hp;
 	SetEntProp(killer, Prop_Data, "m_iHealth", reset_hp);
+	RequestFrame(RegenKiller, killer);
```

Comment:
I'm assuming you've tested this? Not sure if this would work, but if it's really that simple, works for me!

Reply:
https://cdn.discordapp.com/attachments/782024910386102283/782773324723650593/5gezYwB1wi.mp4